### PR TITLE
BUG Refactor HeaderField to be a functional component

### PR DESCRIPTION
**Pull request by maxime-rainville**
The point of this PR is just to force an update to the pattern library so we can test the new set up.

The PR itself:
- Refactors HeaderField to be a functional component
- Adds test to the HeaderField
- Simplify the HeaderField stories to use knobs instead of having a separate entry for each header level

# Parent issue
- https://github.com/silverstripe/silverstripe-admin/issues/1227

Blocked by https://github.com/silverstripe/silverstripe-admin/issues/1215

### CODE REVIEW

**Comment by emteknetnz** [comment]
Failing behat travis failure is unrelated https://github.com/silverstripe/silverstripe-admin/issues/1215

NPM test passed https://travis-ci.com/github/silverstripe/silverstripe-admin/jobs/510361774

File: client/src/components/HeaderField/tests/HeaderField-test.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+/* global jest, describe, it, expect */
+
+import React from 'react';
+import HeaderField from '../HeaderField';
+import Enzyme, { shallow } from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16/build/index';
+
+Enzyme.configure({ adapter: new Adapter() });
+
+describe('Toast', () => {
```
[the remark this answers is not in the document]
Comment (maxime-rainville) [reply client/src/components/HeaderField/tests/HeaderField-test.js]: Fixed

**Review by emteknetnz — APPROVED** [review]
MOG

**Comment by maxime-rainville** [comment]
admin is green ... I've rebased so hopefully the build should be green now :crossed_fingers:

**Pull request merged**